### ISSUE TITLE
Add the current row's information in the onGetErrorMessage callback

### ISSUE THREAD

**Issue by spplante** [SharePoint/sp-dev-fx-property-controls#92]
#### Category
- [x] Enhancement
- [ ] Bug
- [ ] Question

#### Version

1.9.0

#### Expected / Desired Behavior / Question
When using the **onGetErrorMessage** callback on a text field, it would be great to have a second parameter **currentRow: any** in the callback in order to give the user the possibility to do some validations based on the other fields of the current row.

For example, if I have the columns **Home Phone** and **Cell Phone** and I want to do some custom validation on the **Cell Phone** column, like allow empty values only is **Home Phone** is provided.

So instead of having the following callback :

onGetErrorMessage(value: any)

We would have something like this :

onGetErrorMessage(value: any, currentRow: any)

👍

**Comment by estruyf** [comment]
Thanks @spplante for reporting this. I will work on this one to get it included in version `1.10.0`.

**Comment by estruyf** [comment]
I pushed a change for this to support this. The callback provides you with the current value you are editing, index of the row and the item itself with all its properties.

Here is an example of a new item:

![screenshot 2018-09-06 16 30 40](https://user-images.githubusercontent.com/2900833/45164937-f4fa1700-b1f3-11e8-92b4-fb375a90aa18.png)

Here is an example of a stored item:

![screenshot 2018-09-06 16 31 15](https://user-images.githubusercontent.com/2900833/45164990-1b1fb700-b1f4-11e8-8cf9-ba2ea6cacbb6.png)

```
private minLengthValidation (value: string, index: number, item: any) {
  console.log(`Currently editing item nr: ${index === null ? "new item" : index}. It contains the following properties:`, item);
  return value.length >= 3 ? "" : "Should at least contain 3 characters.";
}
```

If you want, you can already test out the latest 1.10.0 beta: https://sharepoint.github.io/sp-dev-fx-property-controls/beta/

**Comment by spplante** [comment]
Good stuff!

**Comment by estruyf** [comment]
Closing the issue as it got released in version `1.10.0`.
